clippy_utils/src/sugg.rs

```diff
@@ -1007,10 +1070,43 @@ mod test {
 
     #[test]
     fn binop_maybe_par() {
-        let sugg = Sugg::BinOp(AssocOp::Add, "(1 + 1)".into());
+        let sugg = Sugg::BinOp(AssocOp::Add, "1".into(), "1".into());
         assert_eq!("(1 + 1)", sugg.maybe_par().to_string());
 
-        let sugg = Sugg::BinOp(AssocOp::Add, "(1 + 1) + (1 + 1)".into());
+        let sugg = Sugg::BinOp(AssocOp::Add, "(1 + 1)".into(), "(1 + 1)".into());
         assert_eq!("((1 + 1) + (1 + 1))", sugg.maybe_par().to_string());
     }
+    #[test]
+    fn not_op() {
+        use AssocOp::{Add, Equal, Greater, GreaterEqual, LAnd, LOr, Less, LessEqual, NotEqual};
+
+        // Invert the comparison operator.
+        let sugg = Sugg::BinOp(Equal, "1".into(), "1".into());
+        assert_eq!("1 != 1", (!sugg).to_string());
```

Comment:
Consider using a helper function here. Also using different left and right values would be a better test.

```rust
test_not(Equal, "x != y");
test_not(Less, "x >= y");
...
```

Reply:
The test has been simplified. thank you!